docs/design/secrets.md

```diff
@@ -0,0 +1,187 @@
+# Secret Distribution
+
+## Abstract
+
+A proposal for the distribution of secrets (passwords, keys, etc) to containers inside Kubernetes
+using a custom volume type.
+
+## Motivation
+
+Secrets are needed in containers to access internal resources like the Kubernetes master or
+external resources such as git repositories, databases, etc. 
+
+A goal of this design is to eliminate or minimize the modifications to containers in order to
+access secrets. Secrets should be placed where the container expects them to be.
+
+## Constraints and Assumptions
+
+* This design does not prescribe a method for storing/transmitting secrets
+* Encryption and node security are orthogonal concerns
+* It is assumed that node and master are secure and that compromising their security could also
```

Comment:
Clarify which secrets a node compromise could expose. I would expect a node has access to the secrets of the containers scheduled on it.